spring-boot-project/spring-boot/src/main/java/org/springframework/boot/json/BasicJsonParser.java

```diff
@@ -70,37 +70,37 @@ private Object parseInternal(int nesting, String json) {
 			return parseMapInternal(nesting + 1, json);
 		}
 		if (json.startsWith("\"")) {
-			return trimTrailingCharacter(trimLeadingCharacter(json, '"'), '"');
+			return trimEdges(json, '"', '"');
 		}
-		try {
-			return Long.valueOf(json);
-		}
-		catch (NumberFormatException ex) {
-			// ignore
-		}
-		try {
-			return Double.valueOf(json);
-		}
-		catch (NumberFormatException ex) {
-			// ignore
-		}
-		return json;
+		return parseNumber(json);
 	}
 
 	private Map<String, Object> parseMapInternal(int nesting, String json) {
 		Map<String, Object> map = new LinkedHashMap<>();
-		json = trimLeadingCharacter(trimTrailingCharacter(json, '}'), '{').trim();
+		json = trimEdges(json, '{', '}').trim();
 		for (String pair : tokenize(json)) {
 			String[] values = StringUtils.trimArrayElements(StringUtils.split(pair, ":"));
 			Assert.state(values[0].startsWith("\"") && values[0].endsWith("\""),
 					"Expecting double-quotes around field names");
-			String key = trimLeadingCharacter(trimTrailingCharacter(values[0], '"'), '"');
+			String key = trimEdges(values[0], '"', '"');
 			Object value = parseInternal(nesting, values[1]);
 			map.put(key, value);
 		}
 		return map;
 	}
 
+	private Object parseNumber(String json) {
```

Comment:
This method could be `static`.

Reply:
Thanks for trying to help, @asashour, but this suggestion doesn't align with our code style where, generally speaking, we only declare methods as static when they're used in a static context. `BasicJsonParser` has no instance variables so, in theory, every method other than those that are being overridden could be declared static but that just makes the code more verbose and adds some visual noise. It would be better leave this as it is.

Reply:
Dear @asashour and @wilkinsona,

Thank you very much for taking the time to review my pull request.

I greatly appreciate your valuable feedback and suggestions!